Switch to GitHub Actions - Part 2 - From within

flip1995 opened the pull request:
This is a continuation of #5071. This time from a branch inside the rust-lang/rust-clippy repo, not from my fork, since secrets are not available in PRs from forks.

Copying the description of #5071 to here:

Closes #4577

~~This is just an experiment. I don't think we have a consensus _if_ we should move away from travis/appveyor.~~ We have consensus: https://github.com/rust-lang/rust-clippy/pull/5071#issuecomment-576647638

~~GHA would let us run up to 20 concurrent jobs. Since we have 15 integration tests and 4 (linux, linux 32-bit, macos, windows) basic tests, we would be able to run everything concurrently.~~ The org has a limit of 60 jobs across the org, so we limit the matrix of the integration tests to 6 concurrent jobs. 

~~Also IIUC we only have to build Clippy once for every initegration test and then only check the repos.~~ Nope, dependent jobs exist, but they won't keep the artifacts (not even the checkout).

TODO before merge:

- [x] Add `DEPLOY_KEY` secret to github repo
- [x] test deployment on test branch `gh-test`#
  - [x] Test normal deployment
  - [x] Test deployment no changes
  - [x] Test deployment of tag
- [x] talk with `@rust-lang/infra` for bors, `@rust-lang/infra` is good with the move (crater also uses GHA+bors)
- [x] ~~Get remark + clippy_dev check to work on pushes (https://github.community/t5/GitHub-Actions/~Builds-are-not-triggered-with-on-paths/m-p/44075; I contacted GH support about this) ~~That seems to start working again yesterday. Let's hope it keeps working.~~ Or not: df9be48. Now it works again: 723786a. I think we should wait, until this is reliable. It appears, that it doesn't work on force pushes (sometimes?): 5814142~~ We need to run the bors tests unconditionally anyway (47138d1) so it doesn't really matter.
- [x] ~~impl homu checks for GHA https://github.com/rust-lang/rust-clippy/pull/5071#issuecomment-576642983 -- I prepared: https://github.com/flip1995/rust-central-station/commit/f40230dc3cebfe4120c9dce027f61c18dd8935cb. I'd suggest to first add GHA and keep the travis and appveyor checks for a few days and to remove them in a second pass. The bors dummy jobs are added in https://github.com/rust-lang/rust-clippy/pull/5088/commits/1a83b7ad7a45aa8db4db7da55a45931e725a1ca3 and work as expected: https://github.com/rust-lang/rust-clippy/pull/5088#issuecomment-582055695. I opened https://github.com/rust-lang/rust-central-station/pull/578~~ See https://github.com/rust-lang/rust-clippy/pull/5088#issuecomment-584581420
- [x] ~Add GHA badge to Cargo.toml (blocked on rust-lang/crates.io # 1838)~ Added a FIXME in 
2332b57
- [x] ~Maybe we should also wait until GHA supports yaml anchors. https://github.community/t5/GitHub-Actions/Support-for-YAML-anchors/td-p/30336/~ WIll probably not be implemented in the near future.
- [x] Add back travis + appveyor files for transition period (!)

changelog: none

flip1995 commented:
I also created this GHA: https://github.com/flip1995/rust-clippy/commit/ab1ef4c409de5b6163be38620b5227d8f31088e9

It automatically tags Clippy 1 day after a rust release. But it's pretty hacky, since it updates itself, after tagging the commit by modifying the GHA yaml file.

flip1995 commented:
`push: paths: ...` seems to be broken: https://github.community/t5/GitHub-Actions/Builds-are-not-triggered-with-on-paths/m-p/44075

phansch commented on .github/workflows/clippy.yml:
```diff
@@ -0,0 +1,101 @@
+      working-directory: clippy_workspace_tests
+    - name: Test clippy-driver
+      run: |
+        (
```
Since it's quite a few lines of code, could this blog be moved to a separate script? That would make editing and reviewing changes easier (syntax highlighting and running `shellsheck` come to mind)

It could probably also be re-used in `clippy_bors.yml`?

flip1995 replied:
Done

phansch commented on .github/workflows/clippy_bors.yml:
```diff
@@ -0,0 +1,232 @@
+        ref: ${{ github.ref }}
+    - name: Check Changelog
+      run: |
+        MESSAGE=$(git log --format=%B -n 1)
```
I think also this block of code would benefit from being in a separate script file

flip1995 replied:
I agree that extracting this in a separate file would be better. Sadly you can only access secrets in the `.yml` file and not in `.sh` files, that are called from the `.yml` file. At least, I don't know how to do this.

phansch replied:
Oh that's a bummer. I don't know too much about GHA, maybe it's possible to export the secret into an ENV var?

flip1995 replied:
I tried that, but then only `***` (or something similar) gets written in the env var, for security reasons.

We could do something like I did in 531ee0a, but in this case it would be pretty much the whole script and in the extracted `.sh` script only the `if ...` would be extracted.

phansch commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,72 @@
+        path: 'out'
+    - name: Deploy
+      run: |
+        set -ex
```
IMO, also this code block should still be in a separate file.

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit ac570f7463deba0e3f6a20103a50944e55fb633d with merge 1db534c6474ffde4915f42ae555c63fbd902607a...
<!-- homu: {"type":"TryBuildStarted","head_sha":"ac570f7463deba0e3f6a20103a50944e55fb633d","merge_sha":"1db534c6474ffde4915f42ae555c63fbd902607a"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit df9be48e4b581678e790c0436071829b39e52789 with merge 1de8087f318803290b25f64ba1ed47fd5ba0c2f6...
<!-- homu: {"type":"TryBuildStarted","head_sha":"df9be48e4b581678e790c0436071829b39e52789","merge_sha":"1de8087f318803290b25f64ba1ed47fd5ba0c2f6"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit df9be48e4b581678e790c0436071829b39e52789 with merge 62f1444995072f3fb5c46807dab0003ce6b0700e...
<!-- homu: {"type":"TryBuildStarted","head_sha":"df9be48e4b581678e790c0436071829b39e52789","merge_sha":"62f1444995072f3fb5c46807dab0003ce6b0700e"} -->

flip1995 commented:
@bors retry

bors commented:
:hourglass: Trying commit df9be48e4b581678e790c0436071829b39e52789 with merge f8c23661721919b18faea95617f6e0ae6556a4ab...
<!-- homu: {"type":"TryBuildStarted","head_sha":"df9be48e4b581678e790c0436071829b39e52789","merge_sha":"f8c23661721919b18faea95617f6e0ae6556a4ab"} -->

bors commented:
:boom: Test timed out
<!-- homu: {"type":"TimedOut"} -->

bors commented:
:umbrella: The latest upstream changes (presumably #5109) made this pull request unmergeable. Please resolve the merge conflicts.

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 723786a07ce8e1cea292ab2beaccd582d8a1e8dd with merge 3cdcf736e1eedaf16c8cc1a1d0a009fefbff4308...
<!-- homu: {"type":"TryBuildStarted","head_sha":"723786a07ce8e1cea292ab2beaccd582d8a1e8dd","merge_sha":"3cdcf736e1eedaf16c8cc1a1d0a009fefbff4308"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 01fbcb9a8a76a30defe74ae8390cad6a054346cb with merge 265d4b972b0f398e83ce1b2d56286feb7ef9a70c...
<!-- homu: {"type":"TryBuildStarted","head_sha":"01fbcb9a8a76a30defe74ae8390cad6a054346cb","merge_sha":"265d4b972b0f398e83ce1b2d56286feb7ef9a70c"} -->

bors commented:
:boom: Test timed out
<!-- homu: {"type":"TimedOut"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit e47dec5f424c3423f61dec96ffdf9f95c1853914 with merge 4d023de18fa5c9e1ec8778ad3f5479b9b09392d9...
<!-- homu: {"type":"TryBuildStarted","head_sha":"e47dec5f424c3423f61dec96ffdf9f95c1853914","merge_sha":"4d023de18fa5c9e1ec8778ad3f5479b9b09392d9"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 81c4c84dc5758c0999b5ec8be2496d556fb7bfaf with merge 34c10a1f128243908c6b7fa2e08f5c74a81d1b15...
<!-- homu: {"type":"TryBuildStarted","head_sha":"81c4c84dc5758c0999b5ec8be2496d556fb7bfaf","merge_sha":"34c10a1f128243908c6b7fa2e08f5c74a81d1b15"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 91d06c3ddd60d3c266a369c27a90cddcc644482f with merge 25b2309f425cde2f39371e9f208b885eb903ea5d...
<!-- homu: {"type":"TryBuildStarted","head_sha":"91d06c3ddd60d3c266a369c27a90cddcc644482f","merge_sha":"25b2309f425cde2f39371e9f208b885eb903ea5d"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit ff4940321892aedec6003db5be14f1279b2cc05d with merge 38189f1f9cb44d4e56297f1a511d23e54bdd3ee4...
<!-- homu: {"type":"TryBuildStarted","head_sha":"ff4940321892aedec6003db5be14f1279b2cc05d","merge_sha":"38189f1f9cb44d4e56297f1a511d23e54bdd3ee4"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 369cfe384cab752424d2033c599c44408b9caee4 with merge 90510f20d5c0012e41e1cd401bf701052e32af5b...
<!-- homu: {"type":"TryBuildStarted","head_sha":"369cfe384cab752424d2033c599c44408b9caee4","merge_sha":"90510f20d5c0012e41e1cd401bf701052e32af5b"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 58693e9b2ad6c65c5160f849bc6f39f332789878 with merge c4f142df76c7d0011d689c091e6916c322e9a3d3...
<!-- homu: {"type":"TryBuildStarted","head_sha":"58693e9b2ad6c65c5160f849bc6f39f332789878","merge_sha":"c4f142df76c7d0011d689c091e6916c322e9a3d3"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 6f68fd800796faa978ca81a8306252bca9d263e5 with merge f107250f42f6130da3fc56f5ac8415a886457fda...
<!-- homu: {"type":"TryBuildStarted","head_sha":"6f68fd800796faa978ca81a8306252bca9d263e5","merge_sha":"f107250f42f6130da3fc56f5ac8415a886457fda"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit dbf82c06ecbca19d21bfede6be3723b11329e9aa with merge 3ecbd54cbad6960e0609cb8ccc28f3d399a43655...
<!-- homu: {"type":"TryBuildStarted","head_sha":"dbf82c06ecbca19d21bfede6be3723b11329e9aa","merge_sha":"3ecbd54cbad6960e0609cb8ccc28f3d399a43655"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 4f328b204c9751cd6180b919764a14d47a5eb8ad with merge 455fe9a8f38eca77c94b802f18417bbd837797c3...
<!-- homu: {"type":"TryBuildStarted","head_sha":"4f328b204c9751cd6180b919764a14d47a5eb8ad","merge_sha":"455fe9a8f38eca77c94b802f18417bbd837797c3"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 8f9f7f6c23549e86c2e0f9ce2142f83587f8916f with merge 2e86f59f2f9728d79a0289eec20879813532d33e...
<!-- homu: {"type":"TryBuildStarted","head_sha":"8f9f7f6c23549e86c2e0f9ce2142f83587f8916f","merge_sha":"2e86f59f2f9728d79a0289eec20879813532d33e"} -->

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 4f328b204c9751cd6180b919764a14d47a5eb8ad with merge 6adc3af248113a36b158749cbaf5d554651aece8...
<!-- homu: {"type":"TryBuildStarted","head_sha":"4f328b204c9751cd6180b919764a14d47a5eb8ad","merge_sha":"6adc3af248113a36b158749cbaf5d554651aece8"} -->

flip1995 commented:
Bors Jobs work like expected:

Successful: 
- test: https://github.com/rust-lang/rust-clippy/runs/426008740
- dev: https://github.com/rust-lang/rust-clippy/runs/425948869
- remark: https://github.com/rust-lang/rust-clippy/runs/425947524

Failing: 
- test: https://github.com/rust-lang/rust-clippy/runs/425939447
- dev: https://github.com/rust-lang/rust-clippy/runs/425935675
- remark: https://github.com/rust-lang/rust-clippy/runs/425934546

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit 3f1af19221d8e9bb16e90a4433c56f40a78b4292 with merge b57d64feb9835c13a542419d872aaae2f9b88ca7...
<!-- homu: {"type":"TryBuildStarted","head_sha":"3f1af19221d8e9bb16e90a4433c56f40a78b4292","merge_sha":"b57d64feb9835c13a542419d872aaae2f9b88ca7"} -->

phansch commented:
> Add GHA badge to Cargo.toml (blocked on rust-lang/crates.io#1838)

I don't think we need to block on this if that PR takes longer to merge. A FIXME in the Cargo.toml would be fine, I think.

> Maybe we should also wait until GHA supports yaml anchors. github.community/t5/GitHub-Actions/Support-for-YAML-anchors/td-p/30336

Reading the thread it sounds like support is not coming anytime soon. I'd be fine merging it, even with the duplication.

flip1995 commented:
>  A FIXME in the Cargo.toml would be fine, I think.

Alright I add a fixme :+1:

> Reading the thread it sounds like support is not coming anytime soon.

Yeah, we should open an issue for this though.

---

How do you feel about keeping the travis/appveyor configuration for a few days vs removing them directly in this PR?

flip1995 commented on .github/workflows/clippy_dev.yml:
```diff
@@ -0,0 +1,66 @@
+    - name: Test update_lints
+      run: cargo dev update_lints --check
+    - name: Test fmt
+      run: cargo dev fmt --check
```
All tests implemented by clippy_dev.

flip1995 commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,31 @@
+      run: |
+        eval "$(ssh-agent -s)"
+        ssh-add - <<< "${{ secrets.DEPLOY_KEY }}"
+        bash .github/deploy.sh
```
See above comments.

flip1995 commented on .remarkrc:
```diff
@@ -0,0 +1,12 @@
+    ["remark-lint-no-literal-urls", false],
+    ["remark-lint-no-shortcut-reference-link", false],
+    ["remark-lint-maximum-line-length", 120]
+  ],
```
Renamed from `.remarkrc.json` and fixed, to support the current remark version.

flip1995 commented on README.md:
```diff
@@ -2,6 +2,7 @@
 
 [![Build Status](https://travis-ci.com/rust-lang/rust-clippy.svg?branch=master)](https://travis-ci.com/rust-lang/rust-clippy)
 [![Windows Build status](https://ci.appveyor.com/api/projects/status/id677xpw1dguo7iw?svg=true)](https://ci.appveyor.com/project/rust-lang-libs/rust-clippy)
+[![Clippy Test](https://github.com/rust-lang/rust-clippy/workflows/Clippy%20Test/badge.svg?branch=master&event=push)](https://github.com/rust-lang/rust-clippy/actions?query=workflow%3A%22Clippy+Test%22+event%3Apush+branch%3Amaster)
```
Will show the badge of the master branch links to the clippy test workflow of the master branch.

flip1995 commented on tests/integration.rs:
```diff
@@ -59,6 +59,10 @@ fn integration_test() {
+    } else if stderr.contains("E0514") {
+        panic!("incompatible crate versions");
+    } else if stderr.contains("failed to run `rustc` to learn about target-specific information") {
+        panic!("couldn't find librustc_driver, consider setting `LD_LIBRARY_PATH`");
```
These errors occured, while figuring out, that `RUSTUP_TOOLCHAIN=master` is required to execute the `integration` test binary successfully. I figured, that it can't harm to fail if these errors should happen when running integration tests.

flip1995 commented on util/export.py:
```diff
@@ -71,7 +71,9 @@ def main():
-        json.dump(list(lints.values()), fp, indent=2)
+        lints = list(lints.values())
+        lints.sort(key=lambda x: x['id'])
+        json.dump(lints, fp, indent=2)
```
Otherwise every master commit, even those that change nothing (rustup, infra, ...) would produce a deployment commit.

phansch commented on .github/driver.sh:
```diff
@@ -0,0 +1,29 @@
+#! /bin/bash
```
```suggestion
#!/bin/bash
```

flip1995 replied:
`shellcheck` complained about `#!/bin/bash`, so I changed it to this. I also was surprised, I'll change it back.

phansch replied:
Huh, I'll need to update my shellcheck then. Do you know the error code? I'd love to learn the reason for that suggestion =)

flip1995 replied:
I just rerun it, and now it didn't complain, weird. I think it suggested me that I should add a shebang with `#!/bin/bash` in the file. Not really sure though.

phansch commented on .github/deploy.sh:
```diff
@@ -1,20 +1,7 @@
-#!/bin/bash
-
-# Automatically deploy on gh-pages
+#! /bin/bash
```
```suggestion
#!/bin/bash
```

phansch commented on .github/workflows/clippy.yml:
```diff
@@ -0,0 +1,83 @@
+    - name: Set LD_LIBRARY_PATH (Linux)
+      run: |
+        SYSROOT=$(rustc --print sysroot)
+        echo "::set-env name=LD_LIBRARY_PATH::${SYSROOT}/lib${LD_LIBRARY_PATH+:${LD_LIBRARY_PATH}}"
```
Don't want to start a formatting bikeshed, but to me yaml steps always read better if they are separated by an empty line each. For comparison:

```yaml
    - name: Build
      run: cargo build --features deny-warnings
    - name: Test
      run: cargo test --features deny-warnings
    - name: Test clippy_lints
      run: cargo test --features deny-warnings
      working-directory: clippy_lints
    - name: Test rustc_tools_util
      run: cargo test --features deny-warnings
      working-directory: rustc_tools_util
```

```yaml
    - name: Build
      run: cargo build --features deny-warnings

    - name: Test
      run: cargo test --features deny-warnings

    - name: Test clippy_lints
      run: cargo test --features deny-warnings
      working-directory: clippy_lints

    - name: Test rustc_tools_util
      run: cargo test --features deny-warnings
      working-directory: rustc_tools_util
```

flip1995 replied:
Hm I grouped it into "Setup"-steps, "Run"-steps, and "post-run"-steps. But I can format it like this and just write `# Setup`, `# Run`, ... above these steps. I'll change it 👍

phansch commented on .github/workflows/clippy.yml:
```diff
@@ -0,0 +1,83 @@
+env:
+  RUST_BACKTRACE: 1
+  CARGO_TARGET_DIR: '${{ github.workspace }}/target'
+  GHA_CI: 1
```
As far as I can tell, this is only used in `tests/fmt.rs`? It should probably be called something like `RUN_FMT` then.

flip1995 commented:
Alright, I applied all review comments and added back the travis+appveyor ci files. I removed the deployment from travis, since the deploy script doesn't gate on the branch anymore, so it would deploy from everywhere.

flip1995 commented:
@bors try

bors commented:
:hourglass: Trying commit de2af6dc652842c45ccab2ce60df7e2717c3cd55 with merge cb7465e77043ceee4becd7ab2342219f508f793d...
<!-- homu: {"type":"TryBuildStarted","head_sha":"de2af6dc652842c45ccab2ce60df7e2717c3cd55","merge_sha":"cb7465e77043ceee4becd7ab2342219f508f793d"} -->

bors commented:
:broken_heart: Test failed - [checks-travis](https://travis-ci.com/rust-lang/rust-clippy/builds/147996688)
<!-- homu: {"type":"TryBuildFailed","builder_url":"https://travis-ci.com/rust-lang/rust-clippy/builds/147996688","builder_name":"checks-travis"} -->

flip1995 commented:
I restarted the travis job of the try build. (Failed because of unicode in the PR body)

bors commented:
:sunny: Try build successful - [checks-travis](https://travis-ci.com/rust-lang/rust-clippy/builds/147996688), [status-appveyor](https://ci.appveyor.com/project/rust-lang-libs/rust-clippy/builds/30654009)
Build commit: cb7465e77043ceee4becd7ab2342219f508f793d (`cb7465e77043ceee4becd7ab2342219f508f793d`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"status-appveyor":"https://ci.appveyor.com/project/rust-lang-libs/rust-clippy/builds/30654009","checks-travis":"https://travis-ci.com/rust-lang/rust-clippy/builds/147996688"},"merge_sha":"cb7465e77043ceee4becd7ab2342219f508f793d"} -->

bors commented:
:umbrella: The latest upstream changes (presumably #5145) made this pull request unmergeable. Please resolve the merge conflicts.

lzutao commented on src/driver.rs:
```diff
@@ -301,22 +312,21 @@ pub fn main() {
-                        })
-                    })
+                    let home = std::env::var("RUSTUP_HOME")
+                        .or_else(|_| std::env::var("MULTIRUST_HOME"))
```
Please use the home crate to use RUSTUP_HOME and CARGO_HOME.

flip1995 replied:
We explicitly only want to get the `rustup_home`, if it is provided by `RUSTUP_HOME`, so trying to get it relative to the home dir (which the `home` crate does), would be wrong here.

lzutao commented on src/driver.rs:
```diff
@@ -328,6 +338,15 @@ pub fn main() {
                 .or_else(|| option_env!("SYSROOT").map(PathBuf::from))
+                .or_else(|| {
+                    let home = option_env!("RUSTUP_HOME")
+                        .or(option_env!("MULTIRUST_HOME"))
```
I don't think we use multirust anymore.

flip1995 replied:
That is for people who compile clippy with a multirust setup, I guess. I'd rather leave this in in this PR and deal with it later, since this is unrelated. This file is only modified, because the move detected this bug.

lzutao commented on src/driver.rs:
```diff
@@ -328,6 +338,15 @@ pub fn main() {
+                    toolchain_path(home, toolchain)
+                })
                 .map(|pb| pb.to_string_lossy().to_string())
                 .expect("need to specify SYSROOT env var during clippy compilation, or use rustup or multirust");
```
```suggestion
                .expect("need to specify SYSROOT env var during clippy compilation, or use rustup");
```

flip1995 replied:
See above. Let's deal with this later.

flip1995 commented:
Alright. I contacted Pietro and he suggested, that we first merge this, without gating bors on GHA. After the test phase of ~1 week (if everything works out) we nuke Travis+AppVeyor and let bors only gate on GHA.

@rust-lang/clippy Any objections on merging this? Otherwise, I'd like to start the GHA test phase.

Manishearth commented:
+1 go ahead when ready

flip1995 commented:
Ok, waiting until tomorrow (12-15h). If there are no objections, I'll merge this.

flip1995 commented:
@bors treeclosed=1000

flip1995 commented:
@bors p=1000

flip1995 commented:
@bors r=Manishearth,flip1995

Let's go :rocket:

bors commented:
:pushpin: Commit a0a21a8752b3e1616d183c9e407e5d78de39bd1e has been approved by `Manishearth,flip1995`

<!-- @bors r=Manishearth,flip1995 a0a21a8752b3e1616d183c9e407e5d78de39bd1e -->
<!-- homu: {"type":"Approved","sha":"a0a21a8752b3e1616d183c9e407e5d78de39bd1e","approver":"Manishearth,flip1995"} -->

bors commented:
:hourglass: Testing commit a0a21a8752b3e1616d183c9e407e5d78de39bd1e with merge fcb231f66117db6876ff264a9b1a53dd0f1681f2...
<!-- homu: {"type":"BuildStarted","head_sha":"a0a21a8752b3e1616d183c9e407e5d78de39bd1e","merge_sha":"fcb231f66117db6876ff264a9b1a53dd0f1681f2"} -->

flip1995 commented:
@bors r=Manishearth,flip1995

bors commented:
:pushpin: Commit 3fac289c3d7f9caea256ce6cc65b54e214aa776e has been approved by `Manishearth,flip1995`

<!-- @bors r=Manishearth,flip1995 3fac289c3d7f9caea256ce6cc65b54e214aa776e -->
<!-- homu: {"type":"Approved","sha":"3fac289c3d7f9caea256ce6cc65b54e214aa776e","approver":"Manishearth,flip1995"} -->

bors commented:
:hourglass: Testing commit 3fac289c3d7f9caea256ce6cc65b54e214aa776e with merge 71a5379bda71f19539dabac0232330561e19be1a...
<!-- homu: {"type":"BuildStarted","head_sha":"3fac289c3d7f9caea256ce6cc65b54e214aa776e","merge_sha":"71a5379bda71f19539dabac0232330561e19be1a"} -->

flip1995 commented:
The `bors test finished` job fail was, because I had to restart the macos check on GHA, because the master toolchain was downloaded with 50kb/s. But then again, I also had to restart the travis macos job because of a timeout.

We should look at the macos job in the next few merges though, since it didn't happen for the first time.

bors commented:
:sunny: Test successful - [checks-travis](https://travis-ci.com/rust-lang/rust-clippy/builds/148540707), [status-appveyor](https://ci.appveyor.com/project/rust-lang-libs/rust-clippy/builds/30743310)
Approved by: Manishearth,flip1995
Pushing 71a5379bda71f19539dabac0232330561e19be1a to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Manishearth,flip1995","base_ref":"master","builders":{"status-appveyor":"https://ci.appveyor.com/project/rust-lang-libs/rust-clippy/builds/30743310","checks-travis":"https://travis-ci.com/rust-lang/rust-clippy/builds/148540707"},"merge_sha":"71a5379bda71f19539dabac0232330561e19be1a"} -->

flip1995 commented:
Deployment works for master pushes: https://github.com/rust-lang/rust-clippy/commit/f0da14e98832e5f12a14a0057d261f50d8d6891b

Releases are hard to test
